[v16]Rend les messages d'erreur plus explicite pour débug

artragis opened the pull request:
Cette PR est là pour pouvoir isoler rapidement le problème derrière le bug  #3328.
En effet jusque là tous nos essais pour comprendre ce bug ont été vains.

Seul problème : comme on avait eu la FBI d'internationaliser les messages d'erreur,
le débug était rendu impossible.

VOilà donc la première pierre pour régler le problème.

@sandhose si tu pouvais déployer cette PR sur la béta + mettre le mode débug, ça serait génial.

Commit:
Model._meta.module_name --> Model._meta.model_name

Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#features-removed-in-1-8

Commit:
Suppression du module inutilisé "django.contrib.sites"

S'il doit revenir, des modifications sont nécessaires.
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#django-contrib-sites

Commit:
Django 1.8: Déclarer le dossier de fixtures par défaut lance une erreur

Donc on supprime cette déclaration qui était de toutes façons inutile même avec Django 1.7.
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#management-commands

Commit:
Suppression d'une relation erronée

Django 1.8 vérifie ces relations et donc lancerait une erreur.
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#select-related-now-checks-given-fields

Commit:
Utilisation d'une API privée qui a changée de package
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#cleanup-of-the-django-template-namespace

Commit:
Passage à Django 1.8 minimal (il reste plein de warnings mais ça marche)

Impacts directs obligatoires :
- Une relation en base change de nom pour être python-compatible (et donc sa migration en base)
- On doit maintenant préciser le paramètre --fake-initial lors d'une migration de MAJ (cf https://docs.djangoproject.com/en/1.8/ref/django-admin/#django-admin-option---fake-initial)
- Debug Toolbar en v1.3
- Fix pour que oauth fonctionne, cf https://github.com/evonove/django-oauth-toolkit/issues/204

Commit:
Une relation N..M nullable n'a pas de sens

Ce qui n'a pas de sens doit disparaître, d'autant plus que ça provoque des warnings.
Cf https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.ManyToManyField.allow_unsaved_instance_assignment

Commit:
RedirectView.permanent va changer de valeur --> on la fixe

Cf https://docs.djangoproject.com/en/1.8/ref/class-based-views/base/#django.views.generic.base.RedirectView.permanent

Commit:
django.utils.tzinfo est obsolète, remplacé par django.utils.timezone

Cf https://docs.djangoproject.com/fr/1.8/ref/utils/#django-utils-tzinfo

artragis committed:
refactor(migration): Supprime les migrations des vieux modules.

artragis committed:
chore(dep): Mise a jour de django-crispy-forms de 1.5.2 a 1.6.0.

artragis committed:
chore(dep): Mise a jour de django de 1.7.10 a 1.8.9.

gustavi committed:
chore(dep): Mise a jour de django-debug-toolbar de 1.3.0 a 1.4.

gustavi committed:
chore(dep): Mise a jour de django-oauth-toolkit de 0.9.0 a 0.10.0.

gustavi committed:
refactor(url): Utilisation nouvelle syntaxe dans les URLs.

- Change le format des urlpatterns.
- Utilise les vues dans la declaration des URLs.
- Correction de tous les reverse.
- Correction de toutes les balises url dans les templates.

gustavi committed:
refactor(1.8): Changement de l'import csrf.

https://docs.djangoproject.com/fr/1.8/ref/templates/api/#django.template.RequestContext

gustavi committed:
refactor(settings): Change la gestion des templates.

gustavi committed:
Correction d'un import

artragis committed:
corrige les erreurs de tutorialv2 et munin

artragis committed:
règle la migration et les avertissements de template du forum

artragis committed:
refactor(form): Retire les warnings sur les forms.

gustavi committed:
chore(dep): Mise a jour de django-munin de 0.2.0 a 0.2.1.

gustavi committed:
Regle le soucis pour Travis et MySQL (merci @DevHugo)

Commit:
Hotfix v15.1-noel

gustavi committed:
Ajout des informations pour la mise en prod de la v17

artragis committed:
Cleanup code and use managers.

Emeric54 committed:
Mail d'inscription

TU

Emeric54 committed:
[ci skip]Met à jour l'année

GerardPaligot committed:
Merge pull request #3326 from Emeric54/fix-year

Emeric54 committed:
[ci skip] Typo page de désinscription

GerardPaligot committed:
fix(fixture): Updates update_index_date field for topics and comments.

GerardPaligot committed:
Merge pull request #3327 from Emeric54/fix-typos

pierre-24 committed:
Merge pull request #3334 from GerardPaligot/fix_3332

Met à jour le paramètre update_index_date pour les sujets et les comments.

GerardPaligot committed:
Merge pull request #3311 from artragis/refactor_and_doc_home_utils

gustavi committed:
Merge pull request #3336 from artragis/fix_3229

[beta v16]fix 3229 dévalidation des tutos

Emeric54 committed:
[ci skip] Typo rejet de la validation

SpaceFox committed:
Merge pull request #3358 from Emeric54/typo-rejectvalidation

[ci skip] Typo rejet de la validation

artragis committed:
make error message bare string

GerardPaligot commented:
Nous sommes d'accord que nous ne mergons pas ça ?

artragis commented:
faites comme vous voulez, mais là le truc corrige des problèmes qui nous 
empêchent de diagnostiquer les causes d'erreur.

Le 11/02/2016 14:50, Gérard Paligot a écrit :

> Nous sommes d'accord que nous ne mergons pas ça ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/3364#issuecomment-182873527.

artragis committed:
pep8

GerardPaligot commented:
Je ne suis pas pour merger des fonctionnalités de debug dans notre dernière release pour tenter de corriger un bug (aussi bloquant soit-il). Par contre, je ne vois aucun problème de tirer la branche sur la bêta pour aider à la correction du bug.

artragis commented:
Ok.

ça ne me dérange pas de faire un cherry-pick par la suite.

Le 11/02/2016 15:05, Gérard Paligot a écrit :

> nière release pour tenter de corriger un bug (aussi bloquant soit-il). 
> Par contre, je ne vois aucun problème de tirer la branche sur la bêta 
> pour aider à la correction du bug.

gustavi commented on zds/tutorialv2/mixins.py:
```diff
@@ -68,9 +68,9 @@ def get_object(self, queryset=None):
                 pk = int(self.request.POST['pk'])
             else:
-                raise Http404(_(u"Impossible de trouver le paramètre 'pk'."))
+                raise Http404(u"Impossible de trouver le paramètre 'pk'.")
```
On peut pas avoir autre chose que « pk » ?

artragis replied:
je n'ai pas modifié les messages, juste leur affichage.

Le 11/02/2016 15:26, Laville Augustin a écrit :

> In zds/tutorialv2/mixins.py 
> https://github.com/zestedesavoir/zds-site/pull/3364#discussion_r52606964:
> 
> > @@ -68,9 +68,9 @@ def get_object(self, queryset=None):
> >              elif 'pk' in self.request.POST:
> >                  pk = int(self.request.POST['pk'])
> >              else:
> > -                raise Http404(_(u"Impossible de trouver le paramètre 'pk'."))
> > -                raise Http404(u"Impossible de trouver le paramètre 'pk'.")
> 
> On peut pas avoir autre chose que « pk » ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/3364/files#r52606964.

gustavi replied:
Je sais ;)

Situphen committed:
Corrige un petite erreur

Situphen committed:
Met à jour la documentation d'installation du front-end (fix #3294)

artragis committed:
améliore le rapport d'erreur

Situphen committed:
Affiche le karma en dessous des avatars (fix #1444)

artragis commented:
Je pense vraiment que le merge est souhaitable.

gustavi commented:
Ça été QA ?

artragis commented:
ça a été pullé sur la béta pour comprendre le bug de firm1.

Commit:
Fix#3366: Ajout de l'extension de l'image

Commit:
Fix#3234: Lien de retour à la liste des sujets

GerardPaligot commented:
Plus nécessaire non ?

GerardPaligot committed:
chore(dep): Mise a jour de pillow de 2.9 a 3.1.1.

Commit:
Finish v16-scorpion

GerardPaligot committed:
chore(dep): Mise a jour de django-filter de 0.9.2 a 0.12.

GerardPaligot committed:
chore(dep): Mise a jour de django_model_utils de 2.4 a 2.4.

GerardPaligot committed:
chore(dep): Mise a jour de django-rest-swagger de 0.2.9 a 0.3..0.

GerardPaligot committed:
chore(dep): Mise a jour de DRF de 3.3.1 a 3.3.2.

Mise a jour de dependances connexes :
- drf-extensions de 0.2.7 a 0.2.8.
- django-cors-middleware de 1.0.0 a 1.2.0.
- Suppression de get_cache()
- Changement de noms de parametres de configuration pour REST_FRAMEWORK

Merci particulier a @DevHugo pour sa contribution.

Commit:
Merge remote-tracking branch 'upstream/release-v16' into dev

GerardPaligot committed:
chore(dep): Mise a jour de coverage de 4.0.1 a 4.0.3.

GerardPaligot committed:
chore(dep): Mise a jour de cairosvg de 1.0.13 a 1.0.19.

GerardPaligot committed:
chore(dep): Mise a jour de beautifulsoup4 de 4.3.2 a 4.4.1.

GerardPaligot committed:
chore(dep): Mise a jour de djangorestframework-xml de 1.0.1 a 1.3.0.

GerardPaligot committed:
chore(dep): Mise a jour de factory-boy de 2.4.1 a 2.6.1.

pierre-24 committed:
Freeze the version of `python-slugify` to 1.1.4 (fix #3383)

(and provide a command to fix broken contents)

pierre-24 committed:
Also deal with contents that would have been created DURING v16.

DevHugo committed:
Correct commit name

Commit:
v16.1-scorpion

SpaceFox committed:
Ajout de notes pour éclaircir le but du fichier

SpaceFox committed:
Merge pull request #3331 from GerardPaligot/django_18

Passage à Django 1.8

gustavi committed:
Merge pull request #3374 from orandin/fix-3366

Correction du bug de la perte de l'extension des images issues d'une archive

gustavi committed:
Merge pull request #3379 from orandin/fix-3234

Lien de retour à la liste des sujets du forum

gustavi committed:
Merge pull request #3371 from Situphen/fix-1444-karma

Affiche le karma en dessous des avatars

gustavi committed:
Merge pull request #3370 from Situphen/doc-front

Met à jour la documentation d'installation du front-end (fix #3294)

gustavi committed:
Merge pull request #3323 from Emeric54/evo-3312

Mail d'inscription

gustavi commented:
On fait quoi ici ?

artragis commented:
je vais refaire ma PR sur master car c'est vraiment nécessaire pour
comprendre ce qu'il se passe.

2016-02-25 9:48 GMT+01:00 Laville Augustin notifications@github.com:

> On fait quoi ici ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3364#issuecomment-188673002
> .

gustavi commented:
Super merci :)
